odl/tomo/geometry/geometry.py

```diff
@@ -164,26 +164,6 @@ def det_refpoint(self, mpar):
             The reference point, an `ndim`-dimensional vector
         """
 
-    @abstractmethod
```

Comment:
> In principle that's true. It would be possible to make the hierarchy a bit more fine-grained (perhaps by duck-typing) by differentiating between geometries where the detector moves relative to the object (CT, SPECT) and ones where it does not (PET, MRI, anything with an array of fixed detectors).

Certainly possible to do and may be a good idea. Having `motion_param` and the things that come along, namely the idea of a detector separate from the rest of the geometry description, `rotation_matrix`, `det_ref_point`, etc is probably confusing in quite some cases.

> The case with the rotation matrix (i.e. a rotating detector in general) could encompass all cases where the detector movement can be parametrised in the in polar coordinates x = r(phi) * d(phi), where r(phi) is the distance from a reference point (e.g. 0) and d(phi) is a direction vector. The circular acquisition curves would be special cases of that, but we also need more general cases e.g. in SPECT, where r(phi) is given as a list of numbers. Probably the detector movement is determined during acquisition so I don't know if the points will lie on any reasonably regular curve.
> The axis-oriented case is similar, except that the polar coordinates only apply perpendicular to the axis, while a third function g(phi) governs the movement along the axis.

This can represent any geometry where a detector with fixed shape moves and rotates according to some parameter in an arbitrary interval product. The current approach assumes that any point on the detector can be given by:

```
point = det_ref_point(motion_param) + rotation_matrix(motion_param).dot(point_in_detector_reference_system(detector_param))
```

This is all kindof how we define the distinction between `motion_params` and `detector_params`. So, imo, as long as we keep motion_params, we should keep the rotation_matrix (and the det_ref_point, since these come in a pair).

> Anyway, this is probably better suited for a separate issue. The removal of the matrix was anyway motivated by the imminent PET geometry implementation, which took a turn towards SPECT in the end, so we can keep the matrix in this PR.

Agree, needs more thought.